Allow user to specify alternative docker executables.

rhatdan opened the pull request:
This will allow us to support docker-latest

rhatdan commented:
@baude,@runcom,@cgwalters PTAL

cgwalters commented on Atomic/util.py:
```diff
@@ -215,3 +215,13 @@ def get_scanners():
     return scanners
+
+def default_docker():
+    atomic_config = get_atomic_config()
```
Shouldn't this be cached?  I.e. it's a once-per-process-start thing?

baude replied:
I like the idea of a self.docker_binary being in the init of atomic.

runcom commented on Atomic/util.py:
```diff
@@ -215,3 +215,13 @@ def get_scanners():
+    default_docker = atomic_config.get('default_docker')
+    if not default_docker:
+        default_docker = "docker"
+    return default_docker
```
should we somehow validate (like in_array(valid_dockers_array_strings) the `default_docker` string got from the config and have a nicer error? not issue otherwise

cgwalters commented:
```
    INSTALL_ARGS = [util.default_docker(), "run",
```

is wrong, we shouldn't be trying to load `/etc/atomic.conf` or really doing anything until the program *really* runs.  I think we'll need to move this out of the constant, and prepend it wherever we reference `INSTALL_ARGS` at runtime.

We keep having problems where simply trying to find the version at build time has side effects like talking to the host's docker daemon.

baude commented on Atomic/util.py:
```diff
@@ -215,3 +215,13 @@ def get_scanners():
+
+def default_docker():
+    atomic_config = get_atomic_config()
+    default_docker = atomic_config.get('default_docker')
```
return atomic_config.get('default_docker', 'docker')

?

rhatdan commented:
Added self.docker_binary()
Removed docker being used in ARGS, and only call when needed.
We could add a verify for the docker binary, but we really want to make sure atomic runs even on a system with no docker client.

rhatdan commented:
I think I answered all of the comments.

rhatdan commented:
@baude,@runcom,@cgwalters PTAL

Also fixed bash completions to use alternate docker command.

runcom commented:
LGTM - just that Atomic/Import.py|Export.py aren't Classes and so we use util directly but I guess this is fine

rhatdan commented:
Yes we should move these to classes @shishir-a412ed

runcom commented:
we could have Export(Atomic) (and Import) so we'll use the binary exe directly configured in the parent class I guess

baude commented:
aside from the other comments, looks good to me dan
